### PR TITLE
dq/controllers

### PR DESCRIPTION
**Pull request by HeyMrQuaidoo**
@eawotwi @dozzlee Kindly Review

### CODE REVIEW

**Comment by HeyMrQuaidoo** [comment]
Note:
-----
Need to re-migrate db to update new user models. To do that:
1) Delete migrations folder and .sqlite3 db in the project folder
2) Run `python manage.py makemigrations` and `python manage.py migrate`
3) Api endpoints http://localhost:8080/redoc http://localhost:8080/#/swagger

File: CodedEvents/backend/fixtures/users.json [hunk 1]
```diff
@@ -0,0 +1,74 @@
+[
```
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/fixtures/users.json]: was using it to pre-fill the db initially will take it out after the PR is approved

File: CodedEvents/backend/managers.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from django.contrib.auth.base_user import BaseUserManager
+from django.utils.translation import ugettext_lazy as _
+
+class CustomUserManager(BaseUserManager):
+    """
+    Custom user model manager where email is the unique identifiers
```
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/managers.py]: email made more sense as we wanted to uniquely identify the user. Didn't want a situation where we uniquely identify users via username and they have 2 different emails associated with one username. Theoretically in the previous model that check might've passed

File: CodedEvents/backend/migrations/0001_initial.py [hunk 3]
```diff
@@ -1,21 +1,56 @@
-# Generated by Django 2.2 on 2019-05-01 05:58
+# Generated by Django 2.1.8 on 2019-05-08 00:25
```
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/migrations/0001_initial.py]: Yeah autogenerated

File: CodedEvents/backend/models.py [hunk 4]
```diff
@@ -1,22 +1,32 @@
...
 class Role(models.Model):
     name = models.CharField(max_length=30)
     desc = models.CharField(max_length=100)
 
 
 # This extend the default User model provided by Django, this allows you to store
 # non-auth related information about a user.
-class Profile(models.Model):
-    user = models.OneToOneField(User, on_delete=models.CASCADE)
+class Profile(AbstractUser):
```
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/models.py]: This overrides the django user model structure. The 1-1 mapping seemed fine however I had an issue when referencing a test users' 'profile' from the django user model class. Overriding the abstract user class gives as more leeway imo

File: CodedEvents/backend/oauth.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
+# # from django.shortcuts import render
+from rest_framework.response import Response
+from rest_framework.decorators import api_view, permission_classes, schema
+from rest_framework.permissions import AllowAny
+import requests
+from django.urls import path
+from django.conf.urls import url
+from .serializers import ProfileCreateSerializer
+import coreapi, coreschema
+from rest_framework.schemas import AutoSchema, ManualSchema
```
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/oauth.py]: no for test purposes planning on removing it. I wanted to discuss our oauth implementation plan before going any further

File: CodedEvents/backend/param_schema.py [hunk 6]
```diff
@@ -0,0 +1,13 @@
+from rest_framework.compat import coreapi, coreschema
```
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/param_schema.py]: This was initially going to be used as a pipe filter for filtering the _doc_ section of the code snippets.  I changed the implementation to just call on the API.core link schema which was much easier and made more sense in function based views as shown below

@schema(AutoSchema(manual_fields=[
    coreapi.Field("token", required=True, location="form", type="string", description="Token to revoke"),
]))

will delete this file

File: CodedEvents/backend/managers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (eawotwi) [reply CodedEvents/backend/managers.py]: yea but sometimes apps have primary email and secondary email. Could make it easy to change emails for the same account?

File: CodedEvents/backend/models.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (eawotwi) [reply CodedEvents/backend/models.py]: How does it give more leeway?

File: CodedEvents/backend/oauth.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (eawotwi) [reply CodedEvents/backend/oauth.py]: okay could you add in a comment that it will be removed later?

**Comment by eawotwi** [comment]
We should probably move faster and improve later, for later PRs could you just add a description detailing the changes you've made

File: CodedEvents/backend/oauth.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/oauth.py]: sure

File: CodedEvents/backend/managers.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (HeyMrQuaidoo) [reply CodedEvents/backend/managers.py]: sure will change the primary id to an account_id that might help

**Pull request merged**